Solutions/CohesitySecurity/Tests/az.test.py

```diff
@@ -1,66 +1,102 @@
+        self.assertNotEqual(self.api_key, "", "api_key is empty")
+
+        # Deploy playbooks
+        bash_command = "./deploy_playbooks.sh"
+        process = subprocess.Popen(bash_command.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
+
+        if output:
+            print("output --> %s" % output.decode())
+
+        if error:
+            print("error --> %s" % error.decode())
+
+    def test_cohesity_close_helios_incident(self):
+        """
+        This test case verifies that the 'Cohesity_Close_Helios_Incident' playbook can successfully close a Helios
+        incident. It first retrieves an incident ID and verifies that the corresponding Helios alert is in the 'kOpen'
+        state. It then runs the playbook and waits for 30 seconds before verifying that the Helios alert is in the
+        'kSuppressed' state.
+        """
+        # self.skipTest("Skipping test_cohesity_close_helios_incident")
+        playbook_name = "Cohesity_Close_Helios_Incident"
+        subscription_id = get_subscription_id()
+        vid, alert_id = get_one_incident_id(self.resource_group, self.workspace_name)
+
+        alert_details = get_alert_details(alert_id, self.api_key)
+        self.assertEqual(alert_details['alertState'], "kOpen")  # maybe we need to close incident after close helios alert, otherwise, this assert might fail.
+        returncode = run_playbook(subscription_id, vid, self.resource_group, self.workspace_name, playbook_name)
+        self.assertEqual(returncode, 0)
+
+        time.sleep(30)  # Sleep for 30 seconds
+
+        alert_details = get_alert_details(alert_id, self.api_key)
+        print("alert_id --> %s" % alert_id)
+        print("api_key --> %s" % self.api_key)
+        self.assertEqual(alert_details['alertState'], "kSuppressed")
+
+    def test_all_incidents_in_helios(self):
+        """
+        This test case verifies that all incidents in the Sentinel workspace have corresponding alerts in Helios, and
+        that the number of alerts in Helios matches the number of incidents in the Sentinel workspace.
+        """
+        ids = get_incident_ids(self.resource_group, self.workspace_name)
+        alert_ids = [alert_id for (vid, alert_id) in ids]
+        for alert_id in alert_ids:
+            self.assertIsNotNone(get_alert_details(alert_id, self.api_key), f"alert_id --> {alert_id} doesn't exist in helios.")
+        alerts_details = get_alerts_details(alert_ids, self.api_key)
+        self.assertEqual(len(alert_ids), len(alerts_details))
+
+    def test_no_dup_incidents(self):
+        """
+        This test case verifies that there are no duplicate incidents in the Sentinel workspace.
+        """
+        ids = get_incident_ids(self.resource_group, self.workspace_name)
+        alert_ids = [alert_id for (vid, alert_id) in ids]
+        return len(alert_ids) != len(np.unique(np.array(alert_ids)))
+
+    def test_alerts_in_sentinel(self):
+        """
```

Comment:
Can you please add log messages for each test mentioning that the test is starting and finished successfully so that someone looking at the output can understand the current status of the test?

Also, in general, since this is testing, it's a good idea to print the output of the commands or the results we get from API responses to aid debugging. When you do this, please ensure that the timestamps are also present in the logs. You can do the latter part in a seperate PR if you wish, but please ensure the first paragraph of my comment is implemented in this PR>

Reply:
the first part is done.

Reply:
also applied this change to another test.py